sense-android-library/src/nl/sense_os/datastorageengine/CommonSenseProxy.java

```diff
@@ -0,0 +1,597 @@
+     * @param theAppKey		An application key that identifies the application to the commonsense server. Cannot be empty.
+     */
+    public CommonSenseProxy (boolean useLiveServer, String theAppKey)
+    {
+        APP_KEY = theAppKey;
+        if(useLiveServer) {
+            URL_BASE = BASE_URL_LIVE;
+            URL_AUTH = BASE_URL_AUTHENTICATION_LIVE;
+        } else {
+            URL_BASE = BASE_URL_STAGING;
+            URL_AUTH = BASE_URL_AUTHENTICATION_STAGING;
+        }
+    }
+
+
+    /**
+     * Login a user
+     *
+     * If a user is already logged in this call will also return the session ID.
+     *
+     * @param username		A user account in commonsense is uniquely identified by a username. Cannot be empty.
+     * @param password		A password in commonsense does not have any specific requirements.
+     *                      It will be MD5 hashed before sending to the server so the user does not have to provide a hashed password. Cannot be empty.
+     * @throws IOException and JSONException
+     * @return				Session ID. Will be null if the call fails.
+     */
+    public String loginUser( String username, String password) throws IOException, JSONException
+    {
+        if(username == null || username.isEmpty() || password == null || password.isEmpty())
+            throw new IOException("invalid input of username or password");
+
+        final String url = URL_AUTH + "/" + URL_LOGIN;
+        final JSONObject user = new JSONObject();
+        user.put("username", username);
+        user.put("password", password);
+
+        Map<String, String> response = request(url, user, null, HTTP_METHOD_POST);
+        // if response code is not 200 (OK), the login was incorrect
+        int result = checkResponseCode(response.get(RESPONSE_CODE), "login");
+
+        String session_id = response.get("session-id");
+
+        if (result != 0 && session_id == null) {
+            // something went horribly wrong
+            Log.w(TAG, "CommonSense login failed: no session id received!");
+            throw new IOException("login failed with no session id");
+        }
+        return session_id;
+    }
+
+
+    /**
+     * Logout the currently logged in user.
+     *
+     * @param sessionID		The sessionID of the user to logout. Cannot be empty.
+     * @throws IOException
+     * @return				Whether or not the logout finished successfully.
+     */
+    public boolean logoutCurrentUser(String sessionID) throws IOException
+    {
+        if(sessionID == null || sessionID.isEmpty())
+            throw new IOException("invalid input of session ID");
+
+        final String url = URL_AUTH + "/" + URL_LOGOUT;
+        Map<String, String> response = request(url, null, sessionID, HTTP_METHOD_POST);
+        // if response code is not 200 (OK), the logout was incorrect
+        String responseCode = response.get(RESPONSE_CODE);
+        int result = checkResponseCode(response.get(RESPONSE_CODE), "logout");
+        if(result != 0)
```

Comment:
That's true. Then for the false condition, if we also want to receive the response code, can I have in log instead of exception ?